src/eager.jl

```diff
@@ -0,0 +1,188 @@
+    op_desc = get_op_def(op.op_name)
+    n_outputs = length(op_desc.output_arg)
+    handles = [TensorHandle() for _ in 1:n_outputs]
+    ptrs = [Ptr{Cvoid}(0) for _ in 1:n_outputs]
+    num_ret = Cint(n_outputs)
+    status = Status()
+    @tfcall(:TFE_Execute, Cvoid, (Ptr{Cvoid}, Ptr{Cvoid}, Ptr{Cint}, Ptr{Cvoid}), op, ptrs, Ref(num_ret), status)
+    check_status(status)
+    for i in 1:n_outputs
+        handles[i].ptr = ptrs[i]
+    end
+    return handles
+end
+
+function test_eager()
+    ctx = EagerContext()
+    h1 = TensorHandle(RawTensor([1,2]))
+    h2 = TensorHandle(RawTensor([3,4]))
+    op = EagerOp(ctx, "Add")
+    add_input(op, h1)
+    add_input(op, h2)
+    dtype = data_type(h1)
+    op["T"] = dtype
+    res = execute(op)
+    return resolve(res[1])
+end
+
+function setindex!(op::EagerOp, tensor::RawTensor, attr_name)
+    status = Status()
+    @tfcall(:TFE_OpSetAttrTensor, Cvoid, (Ptr{Cvoid}, Cstring, Ptr{Cvoid}, Ptr{Cvoid}), op, attr_name, tensor, status)
+    check_status(status)
+end
+
+function setindex!(op::EagerOp, dtype::DataType, attr_name)
+    @tfcall(:TFE_OpSetAttrType, Cvoid, (Ptr{Cvoid}, Cstring, TF_DataType), op, attr_name, dtype|>jl_to_df_type)
+end
+
+function setindex!(op::EagerOp, value::Integer, attr_name)
+    value = Int64(value)
+    @tfcall(:TFE_OpSetAttrInt, Cvoid, (Ptr{Cvoid}, Cstring, Int64), op, attr_name, value)
+end
+
+function setindex!(op::EagerOp, value::Bool, attr_name)
+    @tfcall(:TFE_OpSetAttrBool, Cvoid, (Ptr{Cvoid}, Cstring, Cuchar), op, attr_name, value)
+end
+
+function setindex!(op::EagerOp, value::AbstractFloat, attr_name)
+    value = Float32(value)
+    @tfcall(:TFE_OpSetAttrFloat, Cvoid, (Ptr{Cvoid}, Cstring, Cfloat), op, attr_name, value)
+end
+
+function setindex!(op::EagerOp, value::AbstractString, attr_name)
+    value = String(value)
+    @tfcall(:TFE_OpSetAttrString, Cvoid, (Ptr{Cvoid}, Cstring, Ptr{Cvoid}, Cint), op, attr_name, Vector{UInt8}(value), sizeof(value))
+end
+
+function setindex!(op::EagerOp, value::Vector, attr_name)
+    set_attr_list(op, attr_name, value)
+end
+
+function set_attr_list(op::EagerOp, attr_name, list::Vector{<:Integer})
+    list = Int64[Int64(x) for x in list]
+    @tfcall(:TFE_OpSetAttrIntList, Cvoid, (Ptr{Cvoid}, Cstring, Ptr{Int64}, Cint), op, attr_name, list, length(list))
+end
+
+function set_attr_list(op::EagerOp, attr_name, list::Vector{<:AbstractFloat})
+    list = Float32[Float32(x) for x in list]
+    @tfcall(:TFE_OpSetAttrFloatList, Cvoid, (Ptr{Cvoid}, Cstring, Ptr{Float32}, Cint), op, attr_name, list, length(list))
+end
+
+function set_attr_list(op::EagerOp, attr_name, list::Vector{<:DataType})
+    list = map(jl_to_df_type, list)
+    @tfcall(:TFE_OpSetAttrTypeList, Cvoid, (Ptr{Cvoid}, Cstring, Ptr{Cvoid}, Cint), op, attr_name, list, length(list))
+end
+
+function set_attr_shape_list(op::EagerOp, attr_name, list::Vector)
+    dims = Vector{Int64}[]
+    for shape in list
+        push!(dims, Int64[shape...])
```

Comment:
Isn't thiis an error?

Reply:
I think it's right? At least Keno's PR https://github.com/malmaud/TensorFlow.jl/commit/932547509a9af0f73d7a1831078da0df7bc39d09 touched this code to make it what it is now.